### PR TITLE
Messaging Extension All Command with HTML and WebView rendering

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
We have integrated following ME Action command in existing sample [Messaging Extension Action Commands C#](https://github.com/Trinetra-MSFT/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Controllers/BotController.cs)
1. Adaptive Card in Task Module
2. Task Module Webview
3. Task Module HTML Page

Following commands got added to Manifest as well:
image

Fixes #

## Proposed Changes

  - Integrated adaptive card in Message extension task module 
image
image

  -Added Webview in Task Module on action command
image
image
  - Integrated HTML razor page as Task Module form

image
image

  - Integrated HTML razor view
image

### CODE REVIEW

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TeamsMessagingExtensionsAction.Controllers
+{
+    public class HomeController : Controller
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Controllers/HomeController.cs:9]: Is this controller necessary?   

Can we just use Razor Pages for this?

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Startup.cs [hunk 2]
```diff
@@ -26,7 +26,8 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddControllers().AddNewtonsoftJson();
-            
+            services.AddRazorPages();
+            services.AddMvc();
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Startup.cs:30]: Please review the startup.cs in the TeamsTaskModule sample, which uses Razor Pages

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js [hunk 3]
```diff
@@ -1,45 +1,134 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-const { TeamsActivityHandler, CardFactory } = require('botbuilder');
+const { TeamsActivityHandler, CardFactory, TeamsInfo } = require('botbuilder');
```
Comment (EricDahlvang) [inline samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js:4]: Are these changes intended to be part of this PR?

**Review by EricDahlvang — CHANGES_REQUESTED** [review]

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Controllers/HomeController.cs (hunk 1 above)
Comment (Trinetra-MSFT) [reply samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Controllers/HomeController.cs:9]: @EricDahlvang , We are using the 2 commands HTML Page & MVC WebView, HTML is Razor page which doesn't use any controller (We are rendering it as razor page itself )method but We have integrated this controller for MVC webview

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Startup.cs (hunk 2 above)
Comment (Trinetra-MSFT) [reply samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Startup.cs:30]: Yes done, We have already added razor page service in startup.cs file but we also need MVC service to render Webview as well

File: samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js (hunk 3 above)
Comment (Trinetra-MSFT) [reply samples/javascript_nodejs/51.teams-messaging-extensions-action/bots/teamsMessagingExtensionsActionBot.js]: No Removed this

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs [hunk 4]
```diff
@@ -4,8 +4,10 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Text.Json;
 using System.Threading;
 using System.Threading.Tasks;
+using AdaptiveCards;
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs:10]: AdaptiveCards needs to be added to the .csproj

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs [hunk 5]
```diff
@@ -109,64 +146,327 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
...
+        private MessagingExtensionActionResponse CreateThumbnailCard(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+            CreatThumbnailCardData cardData = JsonConvert.DeserializeObject<CreatThumbnailCardData>(action.Data.ToString());
+            var card = new ThumbnailCard
+            {
+                Title = "ID: " + cardData.EmpId,
+                Subtitle = "Name: " + cardData.EmpName,
+                Text = "E-Mail: " + cardData.EmpEmail,
+                Images = new List<CardImage> { new CardImage("https://3er1viui9wo30pkxh1v2nh4w-wpengine.netdna-ssl.com/wp-content/uploads/prod/2014/10/MSFT_logo_rgb_C-Gray-768x283.png") },
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs:158]: Please host this file in the project, and use IConfiguration to retrieve the project base url

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs [hunk 6]
```diff
@@ -21,15 +23,50 @@ protected override async Task<MessagingExtensionActionResponse> OnTeamsMessaging
...
+            // The user has chosen to create a card by choosing the 'Create Card' context menu command.
+            CreateRazorCardData cardData = JsonConvert.DeserializeObject<CreateRazorCardData>(action.Data.ToString());
+            var card = new HeroCard
+            {
+                Title = "Requested User: " + turnContext.Activity.From.Name,
+                Subtitle = cardData.Title,
+                Text = cardData.DisplayData,
+            };
+
+            var attachments = new List<MessagingExtensionAttachment>();
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs:53]: Please combine this similar code of creating a `new MessagingExtensionAttachment` into a single method, used by all instances.

File: samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs [hunk 7]
```diff
@@ -109,64 +146,327 @@ private MessagingExtensionActionResponse ShareMessageCommand(ITurnContext<IInvok
...
+                    Type = "result",
+                    Attachments = attachments,
+                },
+            };
+        }
+
+        private MessagingExtensionActionResponse CreateAdaptiveCard(ITurnContext<IInvokeActivity> turnContext, MessagingExtensionAction action)
+        {
+            var createCardData = ((JObject)action.Data).ToObject<CreateCardData>();
+            AdaptiveCard adaptiveCard = new AdaptiveCard(new AdaptiveSchemaVersion(1, 0));
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/51.teams-messaging-extensions-action/Bots/TeamsMessagingExtensionsActionBot.cs:212]: Please abstract away the adaptive card creation either into a separate class, or a json conversion. (this method adds 110 lines to this class, which is a bit too much)

**Review by EricDahlvang — CHANGES_REQUESTED** [review]

**Comment by tracyboehrer** [comment]
Closing.  Teams samples are no longer in BF SDK Samples.
